### PR TITLE
Add a `--nosystemd` flag 

### PR DESCRIPTION
**Pull request by vishh**
Setting this flag to true, forces cAdvisor to not assume systemd to be the cgroups owner for docker containers.

#947

### CODE REVIEW

File: container/docker/factory.go [hunk 1]
```diff
@@ -24,7 +24,6 @@ import (
 	"sync"
 
 	"github.com/docker/libcontainer/cgroups"
-	"github.com/fsouza/go-dockerclient"
```
Comment (jimmidyson) [inline container/docker/factory.go]: Doesn't this cause compilation failure for you?

Comment (vishh) [inline container/docker/factory.go]: Yeah. Fixed it now.

**Comment by jimmidyson** [comment]
So the use case for this is systems where systemd is running but docker is not run via systemd?

File: container/docker/factory.go [hunk 2]
```diff
@@ -46,6 +46,8 @@ var dockerRunDir = flag.String("docker_run", "/var/run/docker", "Absolute path t
 // --cgroup-parent have another prefix than 'docker'
 var dockerCgroupRegexp = regexp.MustCompile(`.+-([a-z0-9]{64})\.scope$`)
 
+var noSystemd = flag.Bool("nosystemd", false, "Don't assume systemd is being used by docker.")
```
Comment (jimmidyson) [inline container/docker/factory.go]: How about something like `Explicitly disable  systemd support for Docker cgroups`?

File: container/docker/factory.go [hunk 3]
```diff
@@ -59,17 +61,18 @@ var check = sync.Once{}
 func UseSystemd() bool {
 	check.Do(func() {
 		useSystemd = false
```
Comment (jimmidyson) [inline container/docker/factory.go]: Isn't this a noop consisting when we run this?

File: container/docker/factory.go [hunk 4]
```diff
@@ -59,17 +61,18 @@ var check = sync.Once{}
...
-		for _, cgroupType := range []string{"name=systemd", "cpu"} {
-			mnt, err := cgroups.FindCgroupMountpoint(cgroupType)
-			if err == nil {
-				// systemd presence does not mean systemd controls cgroups.
-				// If system.slice cgroup exists, then systemd is taking control.
-				// This breaks if user creates system.slice manually :)
-				if utils.FileExists(path.Join(mnt, "system.slice")) {
-					useSystemd = true
-					break
+		if !*noSystemd {
```
Comment (jimmidyson) [inline container/docker/factory.go]: Can we just return here rather than inventing the whole next block?

File: container/docker/factory.go (hunk 2 above)
Comment (vishh) [inline container/docker/factory.go]: Done

File: container/docker/factory.go (hunk 4 above)
Comment (vishh) [inline container/docker/factory.go]: Done

File: container/docker/factory.go (hunk 3 above)
Comment (vishh) [inline container/docker/factory.go]: I don't get your comment? Why is it a no-op?

**Comment by vishh** [comment]
@cadvisorJenkinsBot: retest this please

**Comment by vishh** [comment]
@jimmidyson: PTAL.

**Comment by jimmidyson** [comment]
Thanks @vishh. LGTM

**Pull request merged**